Userland/Libraries/LibCore/Environment.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024, Sam Atkins <atkinssj@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Function.h>
+#include <AK/IterationDecision.h>
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+
+namespace Core::Environment {
+
+char** raw_environ();
+
+struct Entry {
+    StringView full_entry;
+    StringView name;
+    StringView value;
+
+    static Entry from_chars(char const* input);
+    static Entry from_string(StringView input);
+};
+
+class EntryIterator {
+public:
+    constexpr bool operator==(EntryIterator other) const { return m_index == other.m_index; }
+    constexpr bool operator!=(EntryIterator other) const { return m_index != other.m_index; }
+
+    constexpr EntryIterator operator++()
+    {
+        ++m_index;
+        return *this;
+    }
+    constexpr EntryIterator operator++(int)
+    {
+        auto result = *this;
+        ++m_index;
+        return result;
+    }
+
+    Entry const operator*() const { return Entry::from_chars(raw_environ()[m_index]); }
```

Comment:
Don't really need the const, but yes, this (non-reference return) addresses my ownership problem.